tlog-tiles.md

```diff
@@ -0,0 +1,163 @@
+
+This document specifies an efficient HTTP API to fetch the signed checkpoint,
+Merkle Tree hashes, and entries of a transparency log containing arbitrary
+entries.
+
+## Conventions used in this document
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this
+document are to be interpreted as described in [BCP 14][] [RFC 2119][] [RFC
+8174][] when, and only when, they appear in all capitals, as shown here.
+
+[RFC 8446]: https://www.rfc-editor.org/rfc/rfc8446.html
+[BCP 14]: https://www.rfc-editor.org/info/bcp14
+[RFC 2119]: https://www.rfc-editor.org/rfc/rfc2119.html
+[RFC 8174]: https://www.rfc-editor.org/rfc/rfc8174.html
+
+## Parameters
+
+A tiled transparency log is defined by a URL *prefix*, a [checkpoint][] origin,
+and one or more Ed25519 [signed note][] public keys.
```

Comment:
I've raised the issue around cryptographic agility before, so this comment might sound familiar, though I've come with concrete examples for the Sigstore ecosystem for why I'd like to see more than just Ed25519.

As a first question, Is it accurate to say that the spec as-is is designed around publicly available logs that will be witnessed by a witness network? Given that the cosignature spec mandates Ed25519, is this the primary motivation for requiring Ed25519 here?

As a compromise, could we say that logs that are to be witnessed by a public witness network MUST use Ed25519, but otherwise logs MAY use any signature algorithm?

If the spec mandates cryptographic agility, it will mean that the log APIs (e.g. configuration) will be designed so that you can provide your own algorithm. If not, then I suspect that Ed25519 will be hardcoded throughout the codebase and make it harder to use alternative algorithms without making significant code changes and maintaining forks. If we go with the suggestion to mandate Ed25519 for public logs but allow for any algorithm for non-public logs, the APIs should be designed around pluggable algorithms and make it much simpler to use other algorithms.

For the Sigstore ecosystem:

As I've mentioned in other issues, the goal is meeting customers where they're at. Last week I met with someone from a package registry who noted that their infrastructure only supports RSA. I'd like to try to nudge them to support more algorithms, but if this is a hard restriction, I'd rather support RSA and have this ecosystem adopt Sigstore. Ed25519 is not nearly as prolific as ECDSA so if I need to convince an ecosystem to make changes to supported algorithms, I believe it'll be easier to convince them to adopt ECDSA (In the previous issues, it was noted there have been vulnerabilities in RSA as a reason to not adopt RSA, but ECDSA has not had the same issues).

Sigstore clients have implemented ECDSA support primarily. Getting all clients onboard with Ed25519 will be another hurdle for adoption of modern logs and further delay our tile adoption. Not infeasible of course, just adds another blocker.

Reply:
Added a carveout for non-public logs. The goal is indeed guaranteeing that witnesses don't need to implement anything else than Ed25519.